### PR TITLE
add mscoco generative benchmark

### PR DESCRIPTION
**Pull request by gpucce**
This PR is supposed to add testing for generative models (e.g. CoCa) https://github.com/mlfoundations/open_clip/pull/308 it is far for complete but for initial testing it should work with `open_clip` 

The command to run it is something like 

`clip_benchmark eval --dataset=mscoco_captions --dataset_root=coco_data_root --task=mscoco_generative --model=coca_ViT-B-32 --output=result.json --batch_size=16 --pretrained=../pretrained_test_model.pt`

### CODE REVIEW

**Comment by rom1504** [comment]
this looks quite simple, but it seems to me there should be captioning task and then make ms coco a dataset

**Comment by gpucce** [comment]
@rom1504 you are absolutely right, if it is ok I would be happier first using this just to run this specific evaluation on coca and once that one is done then add here more tasks or anyway making this more general

**Comment by rom1504** [comment]
can you rebase on main ?

File: clip_benchmark/metrics/mscoco_generative.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+from open_clip.tokenizer import _tokenizer
+
+def evaluate(model, dataloader, batch_size, device, transform, train_dataloader=None, num_workers=None, amp=True, verbose=False):
+    coco = dataloader.dataset.coco
+    indexer = dataloader.dataset.ids
+    results = []
+    for idx, (img, _) in enumerate(tqdm(dataloader)):
+        n_samples = img.shape[0] # for last batch
+        idxs = [indexer[idx * batch_size + id] for id in range(n_samples)]
+        out = model.generate(img.to(device), seq_len=30, generation_type="beam_search",num_beams=6, num_beam_groups=3, sot_token_id=49406, eos_token_id=49407)
```
Comment (rom1504) [inline clip_benchmark/metrics/mscoco_generative.py:14]: is all this still needed @gpucce ?

**Pull request merged**